src/frontend/next/src/config.ts

```diff
@@ -0,0 +1,7 @@
+export default {
+  title: `Telescope`,
+  description: `A tool for tracking blogs in orbit around Seneca's open source involvement`,
+  author: `SDDS Students and professors`,
+  // TODO: we need a way to pass in process.env.API_URL...
```

Comment:
Can we make a .js file in the frontend folder with the `API_URL` in it and just export it? Not very elegant though.
Or something like [this](https://github.com/facebook/create-react-app/issues/4283#issuecomment-380232039), maybe? 

Did Gatsby not have to proxy `.env` variables from backend to frontend?

Reply:
We have our `.env` files in the project root.  Here we define `API_URL`, so the value is known.

From the root, we run the `develop:next` or `build:next` commands, which in turn call `npm run ...` with the appropriate command within the `src/frontend/next` directory.  So step 1 is: get the `env` passed down to the `npm run` process that spans the call to next's dev/build script.

An easier way to do this might be to have the next.js build reach up to the root and load the `.env` config when it begins the build.  I did a version of this, by adding some code to `next.config.js`, which would `require()` the `dotenv` module, then use `process.env` to copy `API_URL` to `NEXT_PUBLIC_API_URL`.  This works, but it's a bit dirty.

Next.js has a built-in way to define `env` files for dev and prod, as well as local overrides.  However, this is separate from our root config, which isn't what we want (i.e., having to configure the app in multiple places is a good way to screw something up).  Not to mention that our Docker config needs to be able to override and pass down various variables in staging and production.

So maybe we have to use my "dirty" solution, and have the next.js build load the top-level config.  If anyone has a cleaner solution, I'm all ears.

Reply:
I've made a new commit to show the method I'm discussing, see https://github.com/Seneca-CDOT/telescope/pull/1555/commits/05801b488b298e26d59c63d5c0c42d698b52c958.  This works.  Is it good enough?